x-pack/plugins/cloud_security_posture/server/create_indices/create_indices.ts

```diff
@@ -242,7 +251,7 @@ const updateIndexTemplate = async (
         aliases: template?.aliases,
       },
       _meta,
-      composed_of: composedOf,
+      composed_of: composedOf.filter((ct) => ct !== STACK_COMPONENT_TEMPLATE_LOGS_SETTINGS),
```

Comment:
@elastic/kibana-cloud-security-posture Not basing the CSP transform index on the `logs@settings` template was necessary due to the `logs@settings` template on serverless specifying a default lifecycle. I could not find a way to override this from the index template, so instead I decided to filter this out for now and just manually specify the settings that were used before.

Reply:
I do think we should remove this special initialization of the transform and index template and use the dedicated feature provided by Fleet for setting up transforms which would not have this issue.

Reply:
@joshdover, you right, we have task for that:
- https://github.com/elastic/kibana/issues/151860